yunwoo/programmers-1/예산.js

```diff
@@ -0,0 +1,35 @@
+function solution(d, budget) {
+  let result = 0;
+  d.sort((a, b) => a - b).reduce((acc, cur) => {
+    if (budget >= acc + cur) {
+      result++;
+    }
+    return acc + cur;
+  }, 0);
+  return result;
+}
+
+function solution(d, budget) {
+  let result = 0;
+  let sumEach = 0;
+  d.sort((a, b) => a - b);
+  for (i = 0; i < d.length; i++) {
+    sumEach += d[i];
+    if (budget >= sumEach) {
+      result++;
+    }
+  }
+  return result;
+}
```

Comment:
```suggestion
function solution(d, budget) {
  let result = 0;
  let sum = 0;
  
  d.sort((a, b) => a - b);
  
  for (let i = 0; i < d.length; i++) {
    sum += d[i];
    
    if (budget < sum) return result;
    result++;
  }
  
  return result;
}
```
- 개인적으로 `주요 문단마다 줄바꿈`을 해주는게 깔끔하다고 느껴지네요!! 예를 들어, 변수 선언 문단, for문, return 문단 👍
- 저는 변수명은 결국 합친다는 의미니까 sum이여도 충분한 것 같습니다.
- for문에서 i `let` 추가해주는거 잊지말아주세요 (스코프 문제 발생)
- 그리고 for문을 돌린다면, `if(budget < sum)`이 된다면 더 이상 순회는 무의미하기에 곧 바로 result를해서 리턴해서 함수를 종료한다면 좀 더 효율적인 코드가 될 것 같습니다

Reply:
- let 또 까먹었네요😂
-  문단나누기, for문 let 붙이기, for문 최적화까지.. 기억하고 적용하 겠습니다 감사합니다🙇‍♂️